clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/AbstractOpenstackLoadBalancerAtomicOperation.groovy

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2016 Target, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.ops.loadbalancer
+
+import com.netflix.spinnaker.clouddriver.openstack.client.BlockingStatusChecker
+import com.netflix.spinnaker.clouddriver.openstack.client.OpenstackClientProvider
+import com.netflix.spinnaker.clouddriver.openstack.config.OpenstackConfigurationProperties
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.servergroup.MemberData
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.servergroup.ServerGroupParameters
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackOperationException
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackProviderException
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackResourceNotFoundException
+import com.netflix.spinnaker.clouddriver.openstack.deploy.ops.StackPoolMemberAware
+import com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup.ServerGroupConstants
+import com.netflix.spinnaker.clouddriver.openstack.domain.LoadBalancerResolver
+import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackCredentials
+import com.netflix.spinnaker.clouddriver.openstack.task.TaskStatusAware
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
+import org.openstack4j.model.heat.Stack
+import org.openstack4j.model.network.ext.LbPoolV2
+import org.openstack4j.model.network.ext.LbProvisioningStatus
+import org.openstack4j.model.network.ext.ListenerV2
+import org.openstack4j.model.network.ext.LoadBalancerV2
+import org.openstack4j.openstack.networking.domain.ext.ListItem
+
+abstract class AbstractOpenstackLoadBalancerAtomicOperation implements TaskStatusAware, StackPoolMemberAware, LoadBalancerResolver {
+
+  OpenstackCredentials openstackCredentials
+
+  AbstractOpenstackLoadBalancerAtomicOperation(OpenstackCredentials openstackCredentials) {
+    this.openstackCredentials = openstackCredentials
+  }
+
+  /**
+   * Update the server group to remove the given load balancer.
+   * @param loadBalancerId
+   */
+  void updateServerGroup(String operation, String region, String loadBalancerId, List<String> loadBalancersToDelete = null) {
```

Comment:
Would the default value of the `loadBalancersToDelete` be better as an empty list?